sweepbatcher/greedy_batch_selection.go

```diff
@@ -0,0 +1,290 @@
+package sweepbatcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/txscript"
+	sweeppkg "github.com/lightninglabs/loop/sweep"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+// greedyAddSweep selects a batch for the sweep using the greedy algorithm,
+// which minimizes costs, and adds the sweep to the batch.
+func (b *Batcher) greedyAddSweep(ctx context.Context, sweep *sweep) error {
```

Comment:
I think some more details about the algorithm in the godoc can be be helpful.

Reply:
Done